functions/Remove-DbaPlanCache.ps1

```diff
@@ -0,0 +1,89 @@
+    
+        .PARAMETER WhatIf
+            If this switch is enabled, no actions are performed but informational messages will be displayed that explain what would happen if the command were to run.
+    
+        .PARAMETER EnableException
+            By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+            This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+            Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+    
+        .NOTES
+            Tags: Memory
+            Author: Tracy Boggiano, databasesuperhero.com
+    
+            dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+            Copyright (C) 2016 Chrissy LeMaire
+            License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+    
+        .LINK
+            https://dbatools.io/Remove-DbaPlanCache
+    
+        .EXAMPLE
+            Remove-DbaPlanCache -SqlInstance sql2017 -Threshold 200
+    
+            Logs into the SQL Server instance "sql2017" and removes plan caches if over 200 MBs.
+    
+        .EXAMPLE
+            Remove-DbaPlanCache -SqlInstance sql2017 -SqlCredential (Get-Credential sqladmin) 
+    
+            Logs into the SQL instance using the SQL Login 'sqladmin' and then Windows instance as 'ad\sqldba'
+            and removes if Threshold over 100 MBs.
+    #>
+        [CmdletBinding(SupportsShouldProcess = $true)]
+        Param (
+            [parameter(Mandatory, ValueFromPipeline)]
+            [Alias("ServerInstance", "SqlServer", "SqlServers")]
+            [DbaInstanceParameter[]]$SqlInstance,
+            [PSCredential]$SqlCredential,
+            [int]$Threshold = 100,
+            [switch]$EnableException
+        )
+        process {
+            foreach ($instance in $SqlInstance) {
+                try {
+                    Write-Message -Level Verbose -Message "Connecting to $instance"
+                    $server = Connect-DbaInstance -SqlInstance $instance -SqlCredential $sqlcredential 
+                }
+                catch {
+                    Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+                }
+    
+                $results = Get-DbaQueryPlan -SqlInstance $instance -SqlCredential $SqlCredential -Threshold $Threshold
+               
+                if ($results.MB -ge $Threshold) {
+                    if ($Pscmdlet.ShouldProcess) {
+                        $server.Query("DBCC FREESYSTEMCACHE('SQL Plans')")
+                        Write-Message -Level Verbose -Message "Plan Cache was cleared on $($server)"
+                    }
+                    else {
+                        Write-Message -Level Verbose -Message "Plan Cache would be cleared on $($server)"
+                    }
```

Comment:
The above can be updated to the following:

```powershell
if ($resulits.MB -ge $Threshold) {
    if ($PsCmdlet.ShouldProcess("$server", "Cleared SQL Plans plan cache")) {
        $server.Query("DBCC FREESYSTEMCACHE('SQL Plans')")
    }
}
else {
    Write-Message -Level Warning -Message "Threshold [$Threshold] below $server plan cache size of [$($results.MB)]"
}
```

If you have a better way to word that warning out by all means...just an example I'm throwing out. Since we are checking against the input for an action we need to provide some message to show that it was not meet, and nothing was done in some manner.